flink-yarn/src/main/java/org/apache/flink/yarn/cli/AbstractYarnCli.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.flink.yarn.cli;
+
+import org.apache.flink.client.cli.AbstractCustomCommandLine;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.DeploymentOptions;
+import org.apache.flink.yarn.configuration.YarnConfigOptions;
+import org.apache.flink.yarn.executors.YarnJobClusterExecutor;
+import org.apache.flink.yarn.executors.YarnSessionClusterExecutor;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+abstract class AbstractYarnCli extends AbstractCustomCommandLine {
+
+	private static final String ID = "yarn-cluster";
+
+	protected Option applicationId =
+		new Option("yid", "yarnapplicationId", true, "Attach to running YARN session");
+
+	protected Option addressOption =
+		new Option("m", "jobmanager", true, "Specify the yarn cluster mode if the argument equals " + ID);
```

Comment:
```suggestion
		new Option("m", "jobmanager", true, "Set to " + ID + " to use YARN execution mode.");
```
This option will be deprecated at some point because we now also have `--target` and the `GenericCli`. It's good to have it like this for now!